### ISSUE TITLE
How to configure this plugin?

### ISSUE THREAD

**Issue by ghillb** [nvim-telescope/telescope-live-grep-args.nvim#9]
Currently there doesn't seem to be a way to pass default parameters. For example one has to always type `--hidden` when searching in dotfiles. Also the rg flag `-.` (short for `--hidden`) does not work.

It would be nice, if this plugin could inherit the `vim_grep_arguments` setting from the default telescope `live_grep` function.

**Comment by alefpereira** [comment]
@ghillb funny thing, I was just [migrating from `builtin.live_grep` to `telescope-rgg.nvim`](https://github.com/alefpereira/dotfiles/commit/ddbcb73586494317ccfbcd6d40e29c80135edbb9) and had the same problem. Couldn't find any docs but taking a look in the code I check there are some ways to do that. You can check the commit if you want, in any case here are some examples:

**TLDR:** One is to simply pass the new command in the option `vimgrep_arguments` like that:
```
:lua require('telescope').extensions.live_grep_raw.live_grep_raw({ vimgrep_arguments = { 'rg', '--hidden'} })
```

You can also create a function for it, and you can custom whatever you want, for example, you can set default values just like that:
```lua
local M = {}
function M.custom_live_grep_raw (opts)
  local defaults = {
    vimgrep_arguments = { 'rg', '--hidden'}
  }
  opts = vim.tbl_extend("force", defaults, opts or {})
  require('telescope').extensions.live_grep_raw.live_grep_raw(opts)
end
return M
```
And then you could call it by running something like:
```
:lua require'my.module'.custom_live_grep_raw()
```

As any other picker, you can also customize even more, like adding `theme`, custom `path_display`, `layout_configs`, and so one:
```lua
function M.live_grep_raw(opts)
  local defaults = {
    vimgrep_arguments = vim.split(M.vimgrep_arguments, ' '),
    path_display = M.my_custom_path_display,
    theme = 'dropdown',
  }
  opts = vim.tbl_extend("force", defaults, opts or {})
  opts = require("telescope.themes")["get_" .. opts.theme](opts)  -- Apply theme
  require("telescope").extensions.live_grep_raw.live_grep_raw(opts)
end
```

I also started a fork changing the plugin to make it possible to configure it through the common extension setup recipe. Soon I'll create a PR for this:
```lua
telescope.setup{
  extensions = {
    live_grep_raw = {
      vimgrep_arguments = {...}
    }
  }
}
telescope.load_extension('live_grep_raw')
```

BR

**Comment by alefpereira** [comment]
Here is the PR #10

**Comment by ghillb** [comment]
@alefpereira
Seems like a solid solution, but I tried both, the mapping:
```
vim.keymap.set({ "n", "v" }, "<c-a-p>", function()
  telescope.extensions.live_grep_raw.live_grep_raw({ vimgrep_arguments = { "rg", "--hidden" } })
end)
```
and the the config with your [branch](https://github.com/alefpereira/telescope-rg.nvim/tree/better-configuration):
```
extensions = {
  live_grep_raw = {
    vimgrep_arguments = { "rg", "--hidden" },
  },
}
```
and both had the following effect:
![Screenshot 2022-02-17 195434](https://user-images.githubusercontent.com/35503959/154552211-04c0bc4f-939c-49a1-9d64-23f48e3d89b7.png)

**Comment by alefpereira** [comment]
I see, this behavior also happens with `builtin.live_grep`. This is because the parser requires some options to correctly parse the `rg` output. The "minimum" recommended command is the following:  

```lua
{
  "rg",
  "--color=never",
  "--no-heading",
  "--with-filename",
  "--line-number",
  "--column",
  "--smart-case"
}
```
       
You can check `vimgrep_arguments` help more details: 
```
:help vimgrep_arguments
```

I'll include this detail about `vimgrep_arguments` in the `README.md` pointing to the help, also going to change the examples to consider this.

**Comment by weeman1337** [comment]
live_grep_args now supports the `additional_args` arg (ha ha) :slightly_smiling_face: See #86 . You can pass args, that are not visible inside the prompt but forwarded to the grep command.
